src/main/starlark/core/compile/cli/BUILD.bazel

```diff
@@ -0,0 +1,40 @@
+load("//src/main/starlark/core/compile:common.bzl", "TYPE")
+load("//src/main/starlark/core/repositories:versions.bzl", "versions")
+load(":toolchain.bzl", "cli_toolchain")
+
+java_import(
+    name = "kotlinc_jar",
+    jars = [
+        "@com_github_jetbrains_kotlin//:kotlin-compiler",
+    ],
+    deps = [],
+)
+
+java_binary(
+    name = "kotlinc",
+    main_class = "org.jetbrains.kotlin.cli.jvm.K2JVMCompiler",
+    runtime_deps = [":kotlinc_jar"],
+)
+
+cli_toolchain(
+    name = "cli_toolchain",
+    api_version = versions.KOTLIN_CURRENT_COMPILER_RELEASE.version,
+    jvm_target = "11",
+    kotlin_stdlibs = [
+        "//kotlin/compiler:annotations",
+        "//kotlin/compiler:kotlin-stdlib",
+        "//kotlin/compiler:kotlin-stdlib-jdk7",
+        "//kotlin/compiler:kotlin-stdlib-jdk8",
+        "//kotlin/compiler:kotlinx-coroutines-core-jvm",
+        "//kotlin/compiler:trove4j",
+    ],
+    kotlinc = ":kotlinc",
+    language_version = versions.KOTLIN_CURRENT_COMPILER_RELEASE.version,
+    zip = "@bazel_tools//tools/zip:zipper",
```

Comment:
I'd just make this the toolchain default instead of making it customizable. Could even make it internal with `_zip`.

Reply:
After getting kicked off using Bazel's jarjar, I'm leaving options open.